pandas/tests/indexes/ranges/test_range.py

```diff
@@ -533,3 +535,24 @@ def test_isin_range(self, base):
         result = base.isin(values)
         expected = np.array([True, False])
         tm.assert_numpy_array_equal(result, expected)
+
+    @pytest.mark.parametrize("copy", [True, False])
+    @pytest.mark.parametrize("name", [None, "foo"])
+    @pytest.mark.parametrize("ordered", [True, False])
+    def test_astype_category(self, copy, name, ordered, simple_index):
+        super().test_astype_category(copy, name, ordered, simple_index)
+
+        # GH 41263
+        idx = simple_index if not name else simple_index.rename(name)
+
+        dtypes = [CategoricalDtype(ordered=ordered)]
+        if ordered is False:
+            # dtype='category' defaults to ordered=False, so only test once
+            dtypes.append("category")
+
+        for dtype in dtypes:
```

Comment:
can you not simply construct the expected result and use `assert_index_equal(exact=True)`

Reply:
Yeah, that's better. It however exposed a bug because `assert_index_equal(exact=True)` didn't actually work as expected.

I fixed that `assert_index_equal` bug and changed the test to use `exact=True`.